internal/controllers/user_devices_controller.go

```diff
@@ -1856,6 +1859,60 @@ func (udc *UserDevicesController) PostMintDevice(c *fiber.Ctx) error {
 	}, sigBytes)
 }
 
+// GetUserDevicesWithCredentials godoc
+// @Description Complete Tesla auth and get devices for authenticated user
+// @Tags        user-devices
+// @Produce     json
+// @Accept      json
+// @Param       user_device body controllers.GetUserDevicesByIntegrationRequest true "all fields are required"
+// @Security    ApiKeyAuth
+// @Success     200 {object} controllers.GetUserDevicesByIntegrationResponse
+// @Security    BearerAuth
+// @Router      /user/devices/{tokenID}/credentials [post]
```

Comment:
Do we ever use `tokenID` here? I feel like something that says "I'm trying to connect to Tesla" should be in the path.